docs/library.md

```diff
@@ -0,0 +1,237 @@
+leaderboards, err := leaderboard.NewClient(host, port, password, db, connectionTimeout)
+```
+
+## Creating or updating member score and retrieving
+
+```
+const leaderboardID = "lbID"
+const playerID = "playerID"
+const score = 100
+const wantToKnowPreviousRank = false //do I want to receive also the previous rank on the user?
+const scoreTTL = "100"               //how many seconds my score will be kept on the leaderboard
+
+member, err := leaderboards.SetMemberScore(context.Background(), leaderboardID, playerID, score, wantToKnowPreviousRank, scoreTTL)
+if err != nil {
+    return err
+}
+
+playerPrinter := func(publicID string, score int64, rank int) {
+    fmt.Printf("Player(id: %s, score: %d rank: %d)\n", publicID, score, rank)
+}
+
+playerPrinter(member.PublicID, member.Score, member.Rank)
+
+const order = "desc"     //if set to asc, will treat the ranking with ascending scores (less is best)
+const includeTTL = false //if set to true, will return the member's score expiration unix timestamp
+member, err = leaderboards.GetMember(context.Background(), leaderboardID, playerID, order, includeTTL)
+if err != nil {
+    return err
+}
+
+playerPrinter(member.PublicID, member.Score, member.Rank)
+```
+
+## Setting and getting member scores in bulk
+
+```
+const leaderboardID = "lbID"
+
+players := leaderboard.Members{
+    &leaderboard.Member{Score: 1000, PublicID: "playerA"},
+    &leaderboard.Member{Score: 2000, PublicID: "playerB"},
+}
+
+err := leaderboards.SetMembersScore(context.Background(), leaderboardID, players, false, "")
+if err != nil {
+    return err
+}
+
+const order = "desc"     //if set to asc, will treat the ranking with ascending scores (less is best)
+const includeTTL = false //if set to true, will return the member's score expiration unix timestamp
+members, err := leaderboards.GetMembers(context.Background(), leaderboardID, []string{"playerA", "playerB"}, order, includeTTL)
+
+for _, member := range members {
+    fmt.Printf("Player(id: %s, score: %d rank: %d)\n", member.PublicID, member.Score, member.Rank)
+}
+```
+
+## Retrieving leaderboard leaders
+
+```
+const leaderboardID = "myleaderboardID"
+
+players := leaderboard.Members{
+    &leaderboard.Member{Score: 10, PublicID: "player1"},
+    &leaderboard.Member{Score: 20, PublicID: "player2"},
+}
+
+err = leaderboards.SetMembersScore(context.Background(), leaderboardID, players, false, "")
+if err != nil {
+    log.Fatalf("leaderboards.SetMembersScore failed: %v", err)
+}
+const pageSize = 10
+const pageIdx = 1 //starts at 1
+leaders, err := leaderboards.GetLeaders(context.Background(), leaderboardID, pageSize, pageIdx, "desc")
+if err != nil {
+    log.Fatalf("leaderboards.GetLeaders failed: %v", err)
+}
+
+for _, player := range leaders {
+    fmt.Printf("Player(id: %s, score: %d rank: %d)\n", player.PublicID, player.Score, player.Rank)
+}
+```
+
+## Incrementing player scores
+
+This code example assumes that there is a playerA score stored on leaderboard lbID
+and  
```

Comment:
I think this sentence was cut short or this `and` was forgotten here.

Reply:
Also, when incrementing it is not required that the score exists previously in the leaderboard

Reply:
ok